web/server/models.py

```diff
@@ -362,3 +369,120 @@ class ArtifactChange(BaseModel):
     path: str
     type: t.Optional[ArtifactType] = None
     file: t.Optional[File] = None
+
+
+class ReportTestsResult(BaseModel):
+    message: str
+
+
+class ReportTestDetails(ReportTestsResult):
+    details: str
+
+
+class ReportTestsFailure(ReportTestsResult):
+    total: int
+    failures: int
+    errors: int
+    successful: int
+    dialect: str
+    details: t.List[ReportTestDetails]
+    traceback: str
+
+
+class BackfillDetails(BaseModel):
+    model_name: t.Optional[str] = None
+    view_name: str
+    interval: t.Tuple[str, str]
+    batches: int
+
+
+class BackfillTask(BaseModel):
+    completed: int
+    total: int
+    view_name: str
+    start: TimeLike
+    end: t.Optional[TimeLike] = None
+
+
+class TrackableMeta(BaseModel):
+    status: Status = Status.init
+    start: int = Field(default_factory=now_timestamp)
+    end: t.Optional[int] = None
+    done: bool = False
+
+    @property
+    def duration(self) -> int | None:
+        return self.end - self.start if self.start and self.end else None
+
+    def dict(self, *args: t.Any, **kwargs: t.Any) -> t.Dict[str, t.Any]:
+        data = super().dict(*args, **kwargs)
+        data["duration"] = self.duration
+        return data
+
+
+class Trackable(BaseModel):
+    meta: TrackableMeta = Field(default_factory=TrackableMeta)
+
+    def stop(self, success: bool = True) -> None:
+        if success:
+            self.meta.status = Status.success
+        else:
+            self.meta.status = Status.fail
+
+        self.meta.end = now_timestamp()
+        self.meta.done = bool(self.meta.start and self.meta.end)
+
+    def update(self, data: t.Dict[str, t.Any]) -> None:
+        for k, v in data.items():
+            setattr(self, k, v)
+
+
+class PlanApplyStageValidation(Trackable):
```

Comment:
What's the difference between PlanApplyStageValidation.start and PlanApplyStageValidation.meta.start?